CHANGELOG.md

```diff
@@ -1,8 +1,75 @@
+* Fix bug in codecs parameters detection in `ALTER MODIFY` of codec without type of column. [#4883](https://github.com/yandex/ClickHouse/pull/4883) ([alesapin](https://github.com/alesapin))
+* Functions cutQueryStringAndFragment() and queryStringAndFragment() now works correctly when url contains a fragment and no query. [#4894](https://github.com/yandex/ClickHouse/pull/4894) ([Vitaly Baranov](https://github.com/vitlibar))
+* Fix bug in `ReadBufferAIO` which was not able to seek backwards after EOF (now it can). In case of concurrent `SELECT` query some threads may "steal" portions of read-data from other threads and seek backwards in buffers. It leads to `Can't adjust last granule` error in rare cases if setting `min_bytes_to_use_direct_io` was set to positive number. [#4897](https://github.com/yandex/ClickHouse/pull/4897) ([alesapin](https://github.com/alesapin))
+* Fix bug with additional file seek and read when block already in uncompressed cached. [#4913](https://github.com/yandex/ClickHouse/pull/4913) ([alesapin](https://github.com/alesapin))
+* Fixed wrong argument types for aggregate functions with low cardinality arguments. https://github.com/yandex/ClickHouse/issues/4919 [#4922](https://github.com/yandex/ClickHouse/pull/4922) ([Nikolai Kochetov](https://github.com/KochetovNicolai))
+* If multiple joins are enabled there were an error on query with JOIN + ARRAY JOIN [#4938](https://github.com/yandex/ClickHouse/pull/4938) ([Artem Zuikov](https://github.com/4ertus2))
+* Fix hanging on start of the server when a dictionary depends on another dictionary via a database with engine=Dictionary. [#4962](https://github.com/yandex/ClickHouse/pull/4962) ([Vitaly Baranov](https://github.com/vitlibar))
+* Set alias for external table (right part of GLOBAL JOIN) to be able to resolve its columns. [#4969](https://github.com/yandex/ClickHouse/pull/4969) ([Artem Zuikov](https://github.com/4ertus2))
+* Partially fix distributed_product_mode=local. It's possible to allow columns of local tables in where/having/order by/... via table aliases. Throw exception if table does not have alias. There's not possible to access to the columns without table aliases yet. [#4986](https://github.com/yandex/ClickHouse/pull/4986) ([Artem Zuikov](https://github.com/4ertus2))
+* Function `toISOWeek` returned wrong results for year 1970. [#4988](https://github.com/yandex/ClickHouse/pull/4988) ([alexey-milovidov](https://github.com/alexey-milovidov))
+* Now follower replicas would not send queries to leader if they receive them from DDL. [#4991](https://github.com/yandex/ClickHouse/pull/4991) ([alesapin](https://github.com/alesapin))
+* Fix wrong result for 'select distinct' with multiple streams (ex. leaded by join) [#5001](https://github.com/yandex/ClickHouse/pull/5001) ([Artem Zuikov](https://github.com/4ertus2))
```

Comment:
Bad style.

Reply:
@4ertus2

Reply:
What's wrong with the style?

Reply:
Maybe 'fix wrong result for select distinct with join'